src/main/java/mate/academy/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+package mate.academy.service;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.OrderItemResponseDto;
+import mate.academy.dto.OrderResponseDto;
+import mate.academy.dto.OrderStatusDto;
+import mate.academy.dto.ShippingAddressRequestDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.mapper.OrderItemMapper;
+import mate.academy.mapper.OrderMapper;
+import mate.academy.model.Order;
+import mate.academy.model.OrderItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.repository.OrderItemRepository;
+import mate.academy.repository.OrderRepository;
+import mate.academy.repository.ShoppingCartRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderItemRepository orderItemRepository;
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public void placeOrder(User user, ShippingAddressRequestDto shippingAddress) {
+        Order order = createOrder(user, shippingAddress);
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
```

Comment:
you call shoppingCartRepository.findShoppingCartByUserId inside createOrder() and instantly after it. Optimize your code to avoid double call of method